butane_cli/src/lib.rs

```diff
@@ -92,6 +93,125 @@ pub fn make_migration(base_dir: &Path, name: Option<&String>) -> Result<()> {
     Ok(())
 }
 
+/// Print a description of a column change indented by two spaces.
+pub fn print_column_diff(old: &AColumn, new: &AColumn) -> Result<()> {
+    if old.typeid()? != new.typeid()? {
+        println!("  type: {:?} -> {:?}", old.typeid()?, new.typeid()?);
+    }
+    if old.is_pk() != new.is_pk() {
+        println!("  pk: {} -> {}", old.is_pk(), new.is_pk());
+    }
+    if old.is_auto() != new.is_auto() {
+        println!("  auto: {} -> {}", old.is_auto(), new.is_auto());
+    }
+    if old.nullable() != new.nullable() {
+        println!("  nullable: {} -> {}", old.nullable(), new.nullable());
+    }
+    if old.unique() != new.unique() {
+        println!("  unique: {} -> {}", old.unique(), new.unique());
+    }
+    if old.default() != new.default() {
+        println!("  default: {:?} -> {:?}", old.default(), new.default());
+    }
+    if old.reference() != new.reference() {
+        let old = match old.reference() {
+            Some(ARef::Literal(reference)) => {
+                format!("{}.{}", reference.table_name(), reference.column_name())
+            }
+            None => "None".to_string(),
+            Some(ARef::Deferred(_)) => panic!(),
+        };
+        let new = match new.reference() {
+            Some(ARef::Literal(reference)) => {
+                format!("{}.{}", reference.table_name(), reference.column_name())
+            }
+            None => "None".to_string(),
+            Some(ARef::Deferred(_)) => panic!(),
+        };
+        println!("  references: {} -> {}", old, new);
+    }
+    Ok(())
+}
+
+/// Print description of a list of [`Operation`].
+pub fn print_ops(ops: Vec<Operation>) -> Result<()> {
+    if ops.is_empty() {
+        println!("No changes");
+        return Ok(());
+    }
+    for op in &ops {
+        use Operation::*;
+        match op {
+            AddTable(table) | AddTableIfNotExists(table) => {
+                println!("New table {}", table.name);
+                for column in &table.columns {
+                    println!("  {} -> {:?}", column.name(), column.typeid()?);
```

Comment:
Consider implementing `Display` for `AColumn` and then just printing the column here.
In the current formatting, I don't understand what the arrow is intended to convey. It seems to have a very different meaning than it does in `print_column_diff`

Reply:
ya, ":" would be better here.
Will implement `Display`.
Also thinking about implementing one of https://crates.io/search?q=indent to do the indentation.

Reply:
I've changed this to `:`.

IMO `Display` isnt appropriate here.  `Display` should display all the details of the column, whereas I want to only summarise it here, so-as to not drown the user with info they already know.  If the user wants all the details, they can (and should) study the SQL.

Reply:
Ok, that's reasonable